File: src/main/java/net/gegy1000/terrarium/server/ServerProxy.java
package net.gegy1000.terrarium.server;

import net.minecraft.world.World;
import net.minecraft.world.WorldServer;
import net.minecraftforge.fml.common.network.simpleimpl.MessageContext;

public class ServerProxy {
    public void onPreInit() {
    }

    public void onInit() {
    }

    public void onPostInit() {
    }

    public boolean hasServer() {
        return true;
    }

    public void openWarnToast(int failCount) {
    }

    public void scheduleTask(MessageContext ctx, Runnable task) {
        WorldServer world = (WorldServer) ctx.getServerHandler().player.world;
        world.addScheduledTask(task);
    }

    public World getWorld(MessageContext ctx) {
        return ctx.getServerHandler().player.world;
    }
}

File: src/main/kotlin/net/gegy1000/terrarium/server/ServerProxy.kt
package net.gegy1000.terrarium.server

import net.gegy1000.terrarium.server.capability.TerrariumCapabilities
import net.gegy1000.terrarium.server.map.source.TerrariumData
import net.gegy1000.terrarium.server.map.source.height.HeightSource
import net.minecraftforge.common.MinecraftForge
import kotlin.concurrent.thread

open class ServerProxy {
    open fun onPreInit() {
        MinecraftForge.EVENT_BUS.register(ServerEventHandler)
        TerrariumCapabilities.onPreInit()

        // TODO: Cache all remote data
        thread(name = "Terrarium Remote Load", start = true, isDaemon = true) {
            TerrariumData.loadInfo()
            HeightSource.loadValidTiles()
        }
    }

    open fun onInit() {
    }

    open fun onPostInit() {
    }
}
